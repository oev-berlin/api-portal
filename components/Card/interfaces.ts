import { ProjectData } from '../../utils/interfaces';

interface CardProps{
    projectData: ProjectData
}

export type { CardProps };
