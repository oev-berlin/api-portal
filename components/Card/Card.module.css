.card {
  background-color: white;
  border-radius: 0.25rem;
  box-shadow: 0 0 1rem 0 rgb(0 0 0 / 12.5%);
  max-width: 350px;
  overflow: hidden;
}

.title {
  padding: 2rem;
  margin: 0;
  text-align: center;
}

.image {
  max-width: 100%;
  height: auto;
  display: block;
}
