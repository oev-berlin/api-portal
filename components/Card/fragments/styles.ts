import {styled} from "@mui/system";
import {Paper} from "@mui/material";

export const MicroserviceContainer = styled(Paper)`
  text-align: center;
  padding: 0.5rem;
`;