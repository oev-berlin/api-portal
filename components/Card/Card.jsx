import React from 'react';
import PropTypes from 'prop-types';
import styles from './Card.module.css';

export const Card = ({ title, imageSrc, imageAlt, ...props }) => (
  <div className={styles.card} {...props}>
    {title.length > 0 && <h3 className={styles.title}>{title}</h3>}
    {imageSrc.length > 0 && (
      <img className={styles.image} src={imageSrc} alt={imageAlt} />
    )}
  </div>
);

Card.propTypes = {
  title: PropTypes.string,
  imageSrc: PropTypes.string,
  imageAlt: PropTypes.string,
};

Card.defaultProps = {
  title: '',
  imageSrc: '',
  imageAlt: '',
};
