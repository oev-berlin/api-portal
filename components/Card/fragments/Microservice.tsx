import React from 'react';
import { Grid } from '@mui/material';
import { MicroserviceContainer } from './styles';
import {MicroServiceProps} from "../interfaces";

export const Microservice = ({ serviceName }: MicroServiceProps) => (
  <Grid item xs={12} sm={4} md={6} lg={4}>
    <MicroserviceContainer>
      {serviceName}
    </MicroserviceContainer>
  </Grid>
);
