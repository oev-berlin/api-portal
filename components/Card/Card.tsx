import {CardActionArea, Grid, Button} from '@mui/material';
import {Microservice} from "./fragments";
import {CardOuterContainer, CardTitle, CardInnerContainer} from "./styles";
import {projectData} from "../../utils/interfaces";

export const Card = ({projectData}: projectData) => (
    <CardOuterContainer>
        <CardActionArea component={Button} sx={{textTransform:"none"}}>
                <CardTitle>
                    {projectData.name}
                </CardTitle>
            <CardInnerContainer container spacing={2}>
                {
                    projectData.microservices.map((microserviceName: string) =>
                        (
                            <Microservice
                                serviceName={microserviceName}
                                key={microserviceName}
                            />
                        )
                    )}
            </CardInnerContainer>
        </CardActionArea>
    </CardOuterContainer>

);