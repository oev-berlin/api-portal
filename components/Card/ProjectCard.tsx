import {CardActionArea, Grid, Button} from '@mui/material';
import {Microservice} from "./fragments";
import {ProjectContainer, ProjectTitle} from "./styles";
import {projectData} from "../../utils/interfaces";

export const ProjectCard = ({projectData}: projectData) => (
    <ProjectContainer>
        <CardActionArea component={Button} sx={{textTransform:"none"}}>
                <ProjectTitle>
                    {projectData.name}
                </ProjectTitle>
            <Grid container spacing={2} sx={{p:1, flexWrap: "wrap"}}>
                {
                    projectData.microservices.map((microserviceName: string) =>
                        (
                            <Microservice
                                serviceName={microserviceName}
                                key={microserviceName}
                            />
                        )
                    )}
            </Grid>
        </CardActionArea>
    </ProjectContainer>

);