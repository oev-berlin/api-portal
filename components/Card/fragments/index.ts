import {Microservice} from "./Microservice";

export {Microservice}