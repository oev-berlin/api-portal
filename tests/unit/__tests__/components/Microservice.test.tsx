import {render, cleanup} from '@testing-library/react';
import {toHaveNoViolations} from 'jest-axe';
import {Microservice} from "../../../../components/Card/fragments";

expect.extend(toHaveNoViolations);

const setupComponent = ({serviceName}: { serviceName: string }) =>
    render(
        <Microservice serviceName={serviceName}/>
    );

describe('Microservice', () => {
    afterEach(() => {
        cleanup();
    });

    describe('UI Tests', () => {
        it('Microservice component (with correct name) should render without crashing', async () => {
            const screen = setupComponent({serviceName: "TestMicroservice"});
            expect(screen.container).toBeDefined();
        });

        it('Microservice component (with empty name) should render without crashing', async () => {
            const screen = setupComponent({serviceName: ""});
            expect(screen.container).toBeDefined();
        });

        it('Microservice component should render correctly', async () => {
            const screen = setupComponent({serviceName: "TestMicroservice"});
            //Check if the Microservice is rendered.
            await screen
                .findByText(/TestMicroservice/)
                .then((element) => expect(element).toBeInTheDocument());
        });
    });

    describe('Snapshots Tests', () => {
        it('should match a basic snapshot (with correct name)', () => {
            const {container} = setupComponent({serviceName: 'TestMicroservice'});
            expect(container).toMatchSnapshot();
        });

        it('should match a basic snapshot (with empty name)', () => {
            const {container} = setupComponent({serviceName: ''});
            expect(container).toMatchSnapshot();
        });
    });
});
