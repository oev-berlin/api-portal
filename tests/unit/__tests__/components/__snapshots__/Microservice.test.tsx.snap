// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Microservice Snapshots Tests should match a basic snapshot (with correct name) 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 styles__MicroserviceContainer-sc-1amr5hl-0 ezHSOK css-1ps6pg7-MuiPaper-root"
    >
      TestMicroservice
    </div>
  </div>
</div>
`;

exports[`Microservice Snapshots Tests should match a basic snapshot (with empty name) 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 styles__MicroserviceContainer-sc-1amr5hl-0 ezHSOK css-1ps6pg7-MuiPaper-root"
    />
  </div>
</div>
`;
